packages/block-library/src/embed/wp-embed-preview.native.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * WordPress dependencies
+ */
+import { memo, useMemo } from '@wordpress/element';
+import { SandBox } from '@wordpress/components';
+
+/**
+ * Checks for WordPress embed events signaling the height change when iframe
+ * content loads or iframe's window is resized.  The event is sent from
+ * WordPress core via the window.postMessage API.
+ *
+ * References:
+ * window.postMessage: https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage
+ * WordPress core embed-template on load: https://github.com/WordPress/WordPress/blob/HEAD/wp-includes/js/wp-embed-template.js#L143
+ * WordPress core embed-template on resize: https://github.com/WordPress/WordPress/blob/HEAD/wp-includes/js/wp-embed-template.js#L187
+ */
+const observeAndResizeJS = `
+    ( function() {
+        if ( ! document.body || ! window.parent ) {
+            return;
+        }
+
+        function sendResize( { data: { secret, message, value } = {} } ) {
+            if (
+                [ secret, message, value ].some(
+                    ( attribute ) => ! attribute
+                ) ||
+                message !== 'height'
+            ) {
+                return;
+            }
+
+            document
+                .querySelectorAll( 'iframe[data-secret="' + secret + '"' )
```

Comment:
I'm not totally sure but after checking [this commit from WordPress repo](https://github.com/WordPress/WordPress/commit/83c3e3e00eee7563a9f25bfba94c28514e8f0435#diff-4d4d84281b7ef33b7527598233466e32e46b4838f44e9e2bc96c9fc95ba6d2daR4-R13), I think it has to do with sandboxing the iframe.

This part is a bit tricky to test because it requires manually editing the script to validate that it actually changes the height, one easy way is to add the followin code `alert('Editing iframe height');` into line 37. If it works, you'll see an alert popup.

Reply:
I as able to get an alert both in initial render and also on device rotation. Thanks for the tip!